shell/platform/linux/fl_mouse_cursor_plugin.cc

```diff
@@ -37,26 +43,22 @@ FlMethodResponse* activate_system_cursor(FlMouseCursorPlugin* self,
   if (fl_value_get_type(kind_value) == FL_VALUE_TYPE_STRING)
     kind = fl_value_get_string(kind_value);
 
-  const gchar* cursor_name = nullptr;
-  if (g_strcmp0(kind, "none") == 0)
-    cursor_name = "none";
-  else if (g_strcmp0(kind, "basic") == 0)
-    cursor_name = "default";
-  else if (g_strcmp0(kind, "click") == 0)
-    cursor_name = "pointer";
-  else if (g_strcmp0(kind, "text") == 0)
-    cursor_name = "text";
-  else if (g_strcmp0(kind, "forbidden") == 0)
-    cursor_name = "not-allowed";
-  else if (g_strcmp0(kind, "grab") == 0)
-    cursor_name = "grab";
-  else if (g_strcmp0(kind, "grabbing") == 0)
-    cursor_name = "grabbing";
-  else if (g_strcmp0(kind, "resizeLeftRight") == 0)
-    cursor_name = "ew-resize";
-  else if (g_strcmp0(kind, "resizeUpDown") == 0)
-    cursor_name = "ns-resize";
-  else
+  if (systemCursorTable == nullptr) {
+    systemCursorTable = g_hash_table_new(g_str_hash, g_str_equal);
+
+    // The following mapping must be kept in sync with Flutter framework's
+    // mouse_cursor.dart
+    define_system_cursor("none", "none");
+    define_system_cursor("click", "pointer");
+    define_system_cursor("text", "text");
+    define_system_cursor("forbidden", "not-allowed");
+    define_system_cursor("grab", "grabbing");
+    define_system_cursor("resizeLeftRight", "ew-resize");
+    define_system_cursor("resizeUpDown", "ns-resize");
+  }
+
+  const gchar* cursor_name = (const gchar*)g_hash_table_lookup(systemCursorTable, kind);
+  if (cursor_name == nullptr) // Including "basic" kind
```

Comment:
Comments should have trailing '.'.

It wasn't immediately obvious to me what this comment was referring to, perhaps it would be simpler to add "basic" into the table. We should probably print a debug message if an unknown cursor is encountered, as that indicates the table is not up to date.